src/EnergyPlus/PipeHeatTransfer.hh

```diff
@@ -311,116 +305,75 @@ namespace PipeHeatTransfer {
 			EnvironmentHeatLossRate( 0.0 ),
 			EnvHeatLossEnergy( 0.0 ),
 			VolumeFlowRate( 0.0 )
-		{}
-
-	};
 
-	// Object Data
-	extern Array1D< PipeHTData > PipeHT;
-	extern Array1D< PipeHeatTransferReport > PipeHTReport;
-
-	// Functions
-
-	void
-	SimPipesHeatTransfer(
-		int const EquipType,
-		std::string const & EquipName, // name of the Pipe Heat Transfer.
-		int & EqNum, // index in local derived types for external calling
-		bool const InitLoopEquip,
-		bool const FirstHVACIteration // component number
-	);
+		{}
 
-	//==============================================================================
+		static
+		std::shared_ptr< PlantComponent > factory( int objectType, std::string objectName );
```

Comment:
This is more than just "create". It is more like "createOrFind".

I guess what is the advantage of storing things in an Array/vector if you just linearly search for the objects anyways? Why not use a `map` or `unordered_map` keyed on the name? The searching would be `O( logn )` and `O( 1 )` respectively AND they both would enforce unique names by default so no need to call VerifyName.

That said, both structures require more memory and can not be accessed by index. Although you seem to be moving away from that approach... Which I don't know if I agree with.